src/quo2/components/wallet/network_routing/view.cljs

```diff
@@ -0,0 +1,197 @@
+                :add-new-timeout    add-new-timeout-fn})))
+         [requesting-data?])
+        [:<>
+         (doall
+          (for [[bar-idx bar] (map-indexed vector network-bars)
+                :let          [bar-max-width      (amount->width (:max-amount bar))
+                               bar-width          (-> (:amount-shared-value bar)
+                                                      (reanimated/get-shared-value)
+                                                      (amount->width))
+                               hide-division?     (or (= last-bar-idx bar-idx) @selecting-network?)
+                               this-bar-selected? (= @selected-network-idx bar-idx)
+                               ;; Side effect: adds this bar's animation to the atom
+                               _ (add-hide-animation bar-idx
+                                                     {:amount->width amount->width
+                                                      :bar           bar})]]
+            ^{:key (str "network-bar-" bar-idx)}
+            [:f> network-bar
+             {:bar           bar
+              :max-width     bar-max-width
+              :total-width   total-width
+              :total-amount  total-amount
+              :bar-division? hide-division?
+              :on-top?       this-bar-selected?
+              :allow-press?  (and (or (not @selecting-network?) this-bar-selected?)
+                                  (not requesting-data?)
+                                  (not @press-locked?))
+              :on-press      #(when-not @selecting-network?
+                                (let [[previous-bars [_ & next-bars]] (split-at bar-idx network-bars)
+                                      number-previous-bars            bar-idx]
+                                  (animation/move-previous-bars
+                                   {:bars                 previous-bars
+                                    :bars-widths-negative bars-widths-negative})
+                                  (animation/move-pressed-bar
+                                   {:bar                  bar
+                                    :bars-widths-negative bars-widths-negative
+                                    :number-previous-bars number-previous-bars})
+                                  (animation/move-next-bars
+                                   {:bars                 next-bars
+                                    :bars-widths-negative bars-widths-negative
+                                    :number-previous-bars (inc number-previous-bars)
+                                    :extra-offset         (- bar-max-width bar-width)
+                                    :add-new-timeout      add-new-timeout-fn}))
+                                (animation/show-max-limit-bar bar-opacity-shared-value)
+                                (reset! selecting-network? true)
+                                (reset! selected-network-idx bar-idx))
```

Comment:
I removed the `selecting-network?` atom :)

I'm glad it was easy to remove since I'm not passing atoms across functions, instead I did a `deref` or pass a fn that mutates the atom. It was pretty easy to see the reach of each atom :)